intlist.c

```diff
@@ -0,0 +1,269 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// intlist.c
+// Program on main branch.
+// ------------------------------------------------------------------------------------------------
+
+
+#include <stdio.h>
+#include <malloc.h>
+#include "intlist.h"
+
+
+struct node {
+   int value;
+   struct node* link;
+};
+
+//Defines head pointer and fixes it to null
+struct node* Create () {
+   struct node* head;
+   printf ("\nAn empty linked list is created \n");
+   head = NULL;
+   return head;
+}
+
+//Prints elements of the list starting from the last element that was added to the list
+void PrintFunc (struct node* s) {
+   struct node* temp = s;
+   if (temp != NULL) {
+      printf ("\nThe list elements are : \n");
+      while (temp != NULL) {
+         printf ("%d -> ", temp->value);
+         temp = temp->link;
+      }
+      printf ("\n");
+   } else { printf ("\nNo elements in the list\n"); }
+}
+
+//Adds element at the end of the list
+void Add (struct node** head, int val) {

```

Comment:
Don't use Node**. It complicates the code and makes it difficult to read and implement.  Have a struct LinkedList {} that holds the pointer to the head and restructure code accordingly.